### ISSUE TITLE
Massive lag in VAB after loading ship with nerv

### ISSUE THREAD

**Issue by cjpeterein** [sswelm/PersistentThrust#4]
[KSP.log.zip](https://github.com/sswelm/PersistentThrust/files/4942369/KSP.log.zip)

#### Steps to reproduce:

1. Install KSP 1.10.0.2917 (LinuxPlayer) en-us
2. Install CKAN v1.27.2
3. Use CKAN to install PersistentThrust v1.7.0.0
4. Launch KSP and start a new sandbox save
5. Enter VAB
6. Create a new ship with mk1 command pod, mk1 liquid fuel fuselage, and a nerv
7. Save the ship
8. Leave VAB
9. Enter VAB (which autoloads the saved ship) -> Experience extreme lag while scrolling and clicking around

#### Workaround:

Remove the nerv engine by dragging to the left.  The lag goes away.  Adding the nerv again and repeating 7, 8 and 9 result in the lag returning.

#### Discussion:

The attached log contains multiple "ArgumentNullException: Value cannot be null" exceptions with PersistentThrust in the stack.  These exceptions start when the ship is loaded (step 9), and end when I remove the nerv engine (the workaround).  The first exception follows.

```
[EXC 14:28:58.680] ArgumentNullException: Value cannot be null.
Parameter name: key
    System.Collections.Generic.Dictionary`2[TKey,TValue].FindEntry (TKey key) (at <ad04dee02e7e4a85a1299c7ee81c79f6>:0)
    System.Collections.Generic.Dictionary`2[TKey,TValue].TryGetValue (TKey key, TValue& value) (at <ad04dee02e7e4a85a1299c7ee81c79f6>:0)
    EffectList.Event (System.String eventName, System.Single power, System.Int32 transformIdx) (at <7b275c7511b34ed2b695641a5b2263f8>:0)
    Part.Effect (System.String effectName, System.Single effectPower, System.Int32 transformIdx) (at <7b275c7511b34ed2b695641a5b2263f8>:0)
    PersistentThrust.PersistentEngine.ApplyEffect (System.String effect, System.Single ratio) (at <dc55571d52ca4cd5ac3eebaaba4d4e6a>:0)
    PersistentThrust.PersistentEngine.UpdateFX () (at <dc55571d52ca4cd5ac3eebaaba4d4e6a>:0)
    PersistentThrust.PersistentEngine.Update () (at <dc55571d52ca4cd5ac3eebaaba4d4e6a>:0)
    UnityEngine.DebugLogHandler:LogException(Exception, Object)
    ModuleManager.UnityLogHandle.InterceptLogHandler:LogException(Exception, Object)
    UnityEngine.Debug:CallOverridenDebugHandler(Exception, Object)
```

**Comment by sswelm** [comment]
I could not reproduce the problem but I think I have identified the problem in the ApplyEffect  which didn't correctly verify the effect string

**Comment by cjpeterein** [comment]
This is fixed in v1.7.2.  Thanks!
